sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/server/SparkSQLOperationManager.scala

```diff
@@ -51,9 +50,6 @@ private[thriftserver] class SparkSQLOperationManager()
     require(sqlContext != null, s"Session handle: ${parentSession.getSessionHandle} has not been" +
       s" initialized or had already closed.")
     val conf = sqlContext.sessionState.conf
-    val hiveSessionState = parentSession.getSessionState
-    setConfMap(conf, hiveSessionState.getOverriddenConfigurations)
-    setConfMap(conf, hiveSessionState.getHiveVariables)
```

Comment:
Based on the definition of https://cwiki.apache.org/confluence/display/Hive/LanguageManual+VariableSubstitution#LanguageManualVariableSubstitution-SubstitutionDuringQueryConstruction , the variable is replaced when building the query. This change could introduce the behavior changes. What is the major motivation to make this change?

Reply:
@gatorsmile Thank you for advice.
In extreme circumstance, i mean if set a lot of hiveconf or hivevar,i think it will reduce performance. And in contrast to replace them when initing session, i think replace them when building query is extra action,we should optimize this.

Reply:
Could you show the perf difference?

Reply:
@gatorsmile sorry, i did not test it's performance. But in code logically, it could have a little in perf and i perform to removing it to SparkSQLSessionManager to handle hiveconf and hivevar.

Reply:
```
cat <<EOF > test.sql
select '\${a}', '\${b}';
set b=MOD_VALUE;
set b;
EOF

beeline -u jdbc:hive2://localhost:10000 --hiveconf a=avalue --hivevar b=bvalue -f test.sql
```
Result:
```
+-----------------+-----------------+--+
|       key       |      value      |
+-----------------+-----------------+--+
| b               | bvalue          |
+-----------------+-----------------+--+
1 row selected (0.022 seconds)
```

It is wrong to set the hivevar/hiveconf variable in every operation, which prevents variable updates.

The intention is just an initialized value, so setting it once in SparkSQLSessionManager#openSession is enough.